appserver/jms/admin/src/main/java/org/glassfish/jms/admin/cli/ChangeMasterBrokerCommand.java

```diff
@@ -259,50 +260,65 @@ public Object run(JmsService param) throws PropertyVetoException, TransactionFai
+            JmsHost tmpJmsHost = (JmsHost) jmsHostsList.get(i);
+            if (tmpJmsHost != null && tmpJmsHost.getName().equals(defaultJmsHostName)) {
+                jmsHost = tmpJmsHost;
             }
-	    return jmsHost;
-      }
-
-     private CompositeData updateMasterBroker(String serverName, String oldMasterBroker, String newMasterBroker) throws Exception {
-        MQJMXConnectorInfo mqInfo = getMQJMXConnectorInfo(serverName, config,serverContext, domain, connectorRuntime);
-
-         //MBeanServerConnection  mbsc = getMBeanServerConnection(tgtName);
-         CompositeData result = null;
-         try {
-             MBeanServerConnection mbsc = mqInfo.getMQMBeanServerConnection();
-             ObjectName on = new ObjectName(
-                     CLUSTER_CONFIG_MBEAN_NAME);
-             Object [] params = null;
-
-             String []  signature = new String [] {
-                      "java.lang.String",
-                      "java.lang.String"};
-                        params = new Object [] {oldMasterBroker, newMasterBroker};
-
-             result = (CompositeData) mbsc.invoke(on, "changeMasterBroker", params, signature);
-         } catch (Exception e) {
-                     logAndHandleException(e, "admin.mbeans.rmb.error_creating_jms_dest");
-         } finally {
-                     try {
-                         if(mqInfo != null) {
-                             mqInfo.closeMQMBeanServerConnection();
-                         }
-                     } catch (Exception e) {
-                       handleException(e);
-                     }
-                 }
-         return result;
-     }
+        }
+        return jmsHost;
+    }
+
+    private CompositeData updateMasterBroker(
+        String serverName, String oldMasterBroker, String newMasterBroker)
+        throws Exception {
+
+        MQJMXConnectorInfo mqInfo = getMQJMXConnectorInfo(
+                                         serverName, config, serverContext,
+                                         domain, connectorRuntime);
+
+        //MBeanServerConnection  mbsc = getMBeanServerConnection(tgtName);
+        MBeanServerConnection mbsc = null;
+        CompositeData result = null;
+        try {
+            try {
+                mbsc = mqInfo.getMQMBeanServerConnection();
+            } catch (Exception e) {
+                String[] params = new String[] {
+                    mqInfo.getBrokerInstanceName(), mqInfo.getASInstanceName()};
+                String emsg = localStrings.getLocalString(
+                    "change.master.broker.cannotConnectOldMasterBroker", 
+                    "Unable to connect to the current master broker {0}.  Most likely reasons: the cluster might not be running, or the server instance {1} associated with the broker {0} might not be running or the broker {0} might not be running.  Please check server logs.", params);
+                logger.log(java.util.logging.Level.WARNING, emsg);
```

Comment:
The updated change has avoided code formatting changes for easy review.  isLoggable() is added for the  WARNING log message.